### PR TITLE
Update GameLog to use `Vec::push`

### LINKED ISSUE #80: Logging with Gamelog is O(N)
Hello!  Thanks for making this tutorial, I'm working through section 1.8 now and I'm really enjoying it so far -- it's fantastic to be able to see and walk-through someone's thought process in system design in an iterative way like this.

I've noticed a few small things that could be improved when going through the tutorial so far and I'd love to help improve them if you're interested in these kinds of changes.

This issue is about the performance of Gamelog, and while it's unlikely to be a bottleneck, it's very easy to drastically improve the performance here!

Code like `gamelog.entries.insert(0, format!("You drink the {}, healing {} hp.", names.get(drink.potion).unwrap().name,  potion.heal_amount),);` requires moving all elements in the vector back by one on every insert, thus O(N) for insert.

Possible solutions:
- Use a [`VecDeque`](https://doc.rust-lang.org/std/collections/struct.VecDeque.html): VecDeques can insert from both the front and the back and should have constant performance here
- Store log messages backwards: use [`Vec::push`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.push) instead of `insert(0, ...)` and use `iter().rev()` to iterate the vector backwards, this should also have constant insertion performance (edit: it's also possible to encapsulate these details by implementing a `log` method on `Gamelog` and implementing an `Iterator` for `Gamelog`, too if you wanted to make incorrect usage impossible)

----

I'd be happy to make a PR updating this, using `sed` or something to update them.  Though now that I'm looking at the repo in more depth, I don't see CI anywhere, would you be interested in having CI set up to ensure all the examples compile before merging pull requests?  Don't let me drag you down a rabbit hole of things that aren't important to you -- but if you are interested I'd love to help out a bit.

### PR DESCRIPTION
**Pull request by MarkMcCaskey**
Resolves #80 

Files mostly edited with perl and sed with a decent amount of trial and error -- I can post the commands I ran or attempt to make a script to make these edits so that the diff is easier to verify.

### CODE REVIEW

File: chapter-70-missiles/src/inventory_system/remove_system.rs [hunk 1]
```diff
@@ -31,4 +31,4 @@ impl<'a> System<'a> for ItemRemoveSystem {
 
         wants_remove.clear();
     }
-}
```
Comment (MarkMcCaskey) [inline chapter-70-missiles/src/inventory_system/remove_system.rs]: Looks like I didn't fix all the missing newline edits -- I'll revisit this tomorrow

**Comment by thebracket** [comment]
Giving this a test locally. It touches a lot of chapters, so it may take a bit. Thanks!

**Comment by MarkMcCaskey** [comment]
Sorry about the extra work!  Let me know if I can help out, I can look into writing a test suite that runs over all the examples on CI after work if you'd like.

**Comment by thebracket** [comment]
No problem on the work - I may have to finish up tomorrow. CI is a great idea, but I should probably pause to think a bit about the best approach. I think there are some free options for Rust CI, I'll need to research a little more; it would be good to get it for both RLTK and the tutorial. I just haven't done it before!

**Comment by thebracket** [comment]
The branch https://github.com/thebracket/rustrogueliketutorial/tree/MarkMcCaskey-master has what I think is a fixed version of this PR in it (the original PR should be pulled/credited). I'm going to test a bit before merging.

**Pull request merged**